build/lin.sh

```diff
@@ -190,101 +231,106 @@ mkdir ${DEPS}/gdkpixbuf
+  --disable-xlib --disable-xcb --disable-win32 --disable-egl --disable-glx --disable-wgl --disable-ps \
+  --disable-trace --disable-interpreter ${LINUX:+--disable-quartz} ${DARWIN:+--enable-quartz-image} \
   LIBS="-lpixman-1 -lfreetype"
 make install-strip
 
 mkdir ${DEPS}/fribidi
 curl -Ls https://github.com/fribidi/fribidi/releases/download/v${VERSION_FRIBIDI}/fribidi-${VERSION_FRIBIDI}.tar.xz | tar xJC ${DEPS}/fribidi --strip-components=1
 cd ${DEPS}/fribidi
-autoreconf -fiv
-./configure --host=${CHOST} --prefix=${TARGET} --enable-shared --disable-static --disable-dependency-tracking
-make install-strip
+# Disable tests
+sed -i'.bak' "/subdir('test')/d" meson.build
+LDFLAGS=${LDFLAGS/\$/} meson setup _build --default-library=static --buildtype=release --strip --prefix=${TARGET} ${MESON} \
+  -Ddocs=false
+ninja -C _build
+ninja -C _build install
 
 mkdir ${DEPS}/pango
 curl -Lks https://download.gnome.org/sources/pango/$(without_patch $VERSION_PANGO)/pango-${VERSION_PANGO}.tar.xz | tar xJC ${DEPS}/pango --strip-components=1
 cd ${DEPS}/pango
 # Disable utils, examples, tests and tools
-sed -i "/subdir('utils')/,+3d" meson.build
-CFLAGS= CXXFLAGS= LDFLAGS=${LDFLAGS/\$/} meson setup _build --buildtype=release --strip --libdir=lib --prefix=${TARGET} ${MESON} \
-  -Dgtk_doc=false -Dintrospection=false
+sed -i'.bak' "/subdir('utils')/{N;N;N;d;}" meson.build
+LDFLAGS=${LDFLAGS/\$/} meson setup _build --default-library=static --buildtype=release --strip --prefix=${TARGET} ${MESON} \
+  -Dgtk_doc=false -Dintrospection=false -Duse_fontconfig=true
 ninja -C _build
 ninja -C _build install
 
 mkdir ${DEPS}/svg
 curl -Lks https://download.gnome.org/sources/librsvg/$(without_patch $VERSION_SVG)/librsvg-${VERSION_SVG}.tar.xz | tar xJC ${DEPS}/svg --strip-components=1
 cd ${DEPS}/svg
-sed -i "s/^\(Requires:.*\)/\1 cairo-gobject pangocairo/" librsvg.pc.in
-# Optimise Rust code for binary size
-sed -i "s/debug = true/debug = false\nopt-level = 's'\nlto = true\ncodegen-units = 1\nincremental = false\npanic = 'abort'/" Cargo.toml
-./configure --host=${CHOST} --prefix=${TARGET} --enable-shared --disable-static --disable-dependency-tracking \
-  --disable-introspection --disable-tools --disable-pixbuf-loader
-make install-strip
-# Clear executable bit from librsvg shared library for WSL support
-execstack -c ${TARGET}/lib/librsvg-2.so || true
+sed -i'.bak' "s/^\(Requires:.*\)/\1 cairo-gobject pangocairo/" librsvg.pc.in
+# Do not include debugging symbols
+sed -i'.bak' "/debug =/ s/= .*/= false/" Cargo.toml
+# LTO optimization does not work for staticlib+rlib compilation
+sed -i'.bak' "s/, \"rlib\"//" librsvg/Cargo.toml
+./configure --host=${CHOST} --prefix=${TARGET} --enable-static --disable-shared --disable-dependency-tracking \
+  --disable-introspection --disable-tools --disable-pixbuf-loader ${DARWIN:+--disable-Bsymbolic}
+make install-strip RUST_TARGET_SUBDIR="${RUST_TARGET}/release"
```

Comment:
Was the intent here to declare an env var for the command?
```diff
- make install-strip RUST_TARGET_SUBDIR="${RUST_TARGET}/release" 
+ RUST_TARGET_SUBDIR="${RUST_TARGET}/release" make install-strip
```

Reply:
Yeah, it's a bit weird. It's intended to overwrite a Makefile variable. `RUST_TARGET_SUBDIR` is only set when cross-compiling.
https://github.com/GNOME/librsvg/blob/3661cee1cd2781cd31ac096be7e157f2e661b36b/configure.ac#L315-L319

Somehow when I enable LTO for all compiler invocations, it fails on `proc-macros`. Passing the `--target` option to Cargo seems to fix this.
https://github.com/kleisauke/libvips-packaging/blob/c22ecff4dc069f4747c6f53f4a6618a25aa59a47/build/lin.sh#L51-L55

Reply:
Got it, thanks, bit of a :exploding_head: moment there.